### ISSUE TITLE
Memory consumption when storing big amount of JsonObjects

### ISSUE THREAD

**Issue by omerb4** [google/gson#1449]
Hi,

I'm working with gson to store a big amount of data while also indexing some of the fields inside.
When I making a HashMap between the element id to the actual JsonObject eventually I get a huge hash map that consume GB's of RAM.
When i'm converting the JsonObject to the actual Java object and then put it inside the HashMap the consumption is reduced significantly.
Is there something I can do or edit in the code to make it save the strings in a better way ?

I saw that:
"Gson doesn't use String.intern() which can cause unnecessary strain on the garbage collector" 

But, I prefer to effect performance but not to harm the RAM usages

**Comment by lyubomyr-shaydariv** [comment]
JSON streaming is probably a way to go for your case.

**Comment by omerb4** [comment]
Yes, I do need. i'm doing on going calculation over time so this data so I need it in memory all the time

**Comment by joerg1985** [comment]
+1 for String.intern() support

This could be enabled/disabled in the GsonBuilder to not harm the performance by default and help in special cases when it is explict enabled.
